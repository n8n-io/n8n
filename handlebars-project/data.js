// Sample data for the form template
const defaultFormData = {
    formTitle: "Contact Us Form",
    formDescription: "Please fill out this form to get in touch with us. We'll respond within 24 hours.",
    formDescriptionMetadata: "Contact form for customer inquiries and support requests",
    buttonLabel: "Submit Form",
    testRun: true,
    appendAttribution: true,
    n8nWebsiteLink: "https://n8n.io",
    useResponseData: false,
    
    // Sample form fields demonstrating different input types
    formFields: [
        {
            id: "fullName",
            label: "Full Name",
            type: "text",
            placeholder: "Enter your full name",
            defaultValue: "",
            isInput: true,
            inputRequired: "form-required",
            errorId: "error-fullName"
        },
        {
            id: "email",
            label: "Email Address",
            type: "email",
            placeholder: "Enter your email address",
            defaultValue: "",
            isInput: true,
            inputRequired: "form-required",
            errorId: "error-email"
        },
        {
            id: "phone",
            label: "Phone Number",
            type: "tel",
            placeholder: "Enter your phone number",
            defaultValue: "",
            isInput: true,
            inputRequired: "",
            errorId: "error-phone"
        },
        {
            id: "message",
            label: "Message",
            placeholder: "Tell us how we can help you...",
            defaultValue: "",
            isTextarea: true,
            inputRequired: "form-required",
            errorId: "error-message"
        },
        {
            id: "priority",
            label: "Priority Level",
            isSelect: true,
            selectOptions: ["Low", "Medium", "High", "Urgent"],
            inputRequired: "form-required",
            errorId: "error-priority"
        },
        {
            id: "contactMethod",
            label: "Preferred Contact Method",
            isMultiSelect: true,
            radioSelect: true,
            inputRequired: "form-required",
            errorId: "error-contactMethod",
            multiSelectOptions: [
                { id: "contact-email", label: "Email" },
                { id: "contact-phone", label: "Phone" },
                { id: "contact-sms", label: "SMS" }
            ]
        },
        {
            id: "services",
            label: "Services Interested In",
            isMultiSelect: true,
            radioSelect: false,
            minSelectedOptions: 1,
            maxSelectedOptions: 3,
            inputRequired: "",
            errorId: "error-services",
            multiSelectOptions: [
                { id: "service-web", label: "Web Development" },
                { id: "service-mobile", label: "Mobile Apps" },
                { id: "service-consulting", label: "Consulting" },
                { id: "service-support", label: "Technical Support" }
            ]
        },
        {
            id: "attachment",
            label: "Attach File (Optional)",
            isFileInput: true,
            acceptFileTypes: ".pdf,.doc,.docx,.txt",
            multipleFiles: "",
            placeholder: "Choose file...",
            inputRequired: "",
            errorId: "error-attachment"
        },
        {
            id: "termsAccepted",
            label: "I agree to the terms and conditions",
            isMultiSelect: true,
            radioSelect: false,
            exactSelectedOptions: 1,
            inputRequired: "form-required",
            errorId: "error-termsAccepted",
            multiSelectOptions: [
                { id: "terms-checkbox", label: "Yes, I agree to the terms and conditions" }
            ]
        },
        {
            id: "welcomeMessage",
            isHtml: true,
            html: "<h3>Welcome!</h3><p>Thank you for your interest in our services. Please fill out the form below and we'll get back to you as soon as possible.</p><ul><li>Response time: 24 hours</li><li>Available Monday-Friday</li><li>Emergency support available</li></ul>"
        },
        {
            id: "hiddenField",
            isHidden: true,
            hiddenValue: "contact-form-v1.2"
        }
    ],
    
    formSubmittedHeader: "Thank You!",
    formSubmittedText: "Your message has been received. We'll get back to you within 24 hours.",
    
    // Custom CSS for styling (optional)
    dangerousCustomCss: `
        .container {
            max-width: 600px;
        }
        .form-header h1 {
            color: #2c3e50;
        }
        .form-header p {
            color: #7f8c8d;
        }
    `
};

// Make data available globally
window.defaultFormData = defaultFormData;
