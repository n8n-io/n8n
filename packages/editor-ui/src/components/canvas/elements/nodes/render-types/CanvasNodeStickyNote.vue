<script setup lang="ts">
/* eslint-disable vue/no-multiple-template-root */
import { useCanvasNode } from '@/composables/useCanvasNode';
import type { CanvasNodeStickyNoteRender } from '@/types';
import { ref, computed } from 'vue';
import { NodeResizer } from '@vue-flow/node-resizer';
import type { OnResize } from '@vue-flow/node-resizer/dist/types';
import type { XYPosition } from '@vue-flow/core';

const emit = defineEmits<{
	update: [parameters: Record<string, unknown>];
	move: [position: XYPosition];
	dblclick: [event: MouseEvent];
}>();

const { id, render } = useCanvasNode();

const renderOptions = computed(() => render.value.options as CanvasNodeStickyNoteRender['options']);

/**
 * Resizing
 */

function onResize(event: OnResize) {
	emit('move', {
		x: event.params.x,
		y: event.params.y,
	});

	emit('update', {
		...(event.params.width ? { width: event.params.width } : {}),
		...(event.params.height ? { height: event.params.height } : {}),
	});
}

/**
 * Content change
 */

const isActive = ref(false);

function onInputChange(value: string) {
	emit('update', {
		content: value,
	});
}

function onEdit(edit: boolean) {
	isActive.value = edit;
}

function onDoubleClick(event: MouseEvent) {
	emit('dblclick', event);
}
</script>
<template>
	<NodeResizer
		:min-height="80"
		:min-width="150"
		:height="renderOptions.height"
		:width="renderOptions.width"
		@resize="onResize"
	/>
	<N8nSticky
		:id="id"
		data-test-id="canvas-sticky-note-node"
		:height="renderOptions.height"
		:width="renderOptions.width"
		:class="$style.sticky"
		:model-value="renderOptions.content"
		:background="renderOptions.color"
		:edit-mode="isActive"
		@edit="onEdit"
		@dblclick="onDoubleClick"
		@update:model-value="onInputChange"
	/>
</template>

<style lang="scss" module>
.sticky {
	position: relative;
}
</style>
