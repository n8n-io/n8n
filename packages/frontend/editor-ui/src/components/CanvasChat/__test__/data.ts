import { createTestNode, createTestWorkflow, mockNodeTypeDescription } from '@/__tests__/mocks';
import {
	AGENT_NODE_TYPE,
	AI_CATEGORY_AGENTS,
	AI_SUBCATEGORY,
	CHAT_TRIGGER_NODE_TYPE,
	MANUAL_TRIGGER_NODE_TYPE,
} from '@/constants';
import { type IExecutionResponse } from '@/Interface';
import { WorkflowOperationError } from 'n8n-workflow';

export const nodeTypes = [
	mockNodeTypeDescription({
		name: CHAT_TRIGGER_NODE_TYPE,
		version: 1,
		group: ['trigger'],
	}),
	mockNodeTypeDescription({
		name: MANUAL_TRIGGER_NODE_TYPE,
		version: 1,
		group: ['trigger'],
	}),
	mockNodeTypeDescription({
		name: AGENT_NODE_TYPE,
		codex: {
			subcategories: {
				[AI_SUBCATEGORY]: [AI_CATEGORY_AGENTS],
			},
		},
		version: 1,
	}),
];

export const chatTriggerNode = createTestNode({ name: 'Chat', type: CHAT_TRIGGER_NODE_TYPE });
export const manualTriggerNode = createTestNode({ name: 'Manual' });
export const aiAgentNode = createTestNode({ name: 'AI Agent', type: AGENT_NODE_TYPE });
export const aiModelNode = createTestNode({ name: 'AI Model' });

export const aiManualWorkflow = createTestWorkflow({
	nodes: [manualTriggerNode, aiAgentNode, aiModelNode],
	connections: {
		Manual: {
			main: [[{ node: 'AI Agent', index: 0, type: 'main' }]],
		},
		'AI Model': {
			ai_languageModel: [[{ node: 'AI Agent', index: 0, type: 'ai_languageModel' }]],
		},
	},
});

export const aiChatWorkflow = createTestWorkflow({
	nodes: [chatTriggerNode, aiAgentNode, aiModelNode],
	connections: {
		Chat: {
			main: [[{ node: 'AI Agent', index: 0, type: 'main' }]],
		},
		'AI Model': {
			ai_languageModel: [[{ node: 'AI Agent', index: 0, type: 'ai_languageModel' }]],
		},
	},
});

export const aiChatExecutionResponse: IExecutionResponse = {
	id: 'test-exec-id',
	finished: true,
	mode: 'manual',
	status: 'success',
	data: {
		resultData: {
			runData: {
				'AI Agent': [
					{
						executionStatus: 'success',
						startTime: +new Date('2025-03-26T00:00:00.002Z'),
						executionTime: 1778,
						source: [],
						data: {},
					},
				],
				'AI Model': [
					{
						executionStatus: 'error',
						startTime: +new Date('2025-03-26T00:00:00.003Z'),
						executionTime: 1777,
						source: [],
						error: new WorkflowOperationError('Test error', aiModelNode, 'Test error description'),
						data: {
							ai_languageModel: [
								[
									{
										json: {
											tokenUsage: {
												completionTokens: 222,
												promptTokens: 333,
												totalTokens: 555,
											},
										},
									},
								],
							],
						},
					},
				],
			},
		},
	},
	workflowData: aiChatWorkflow,
	createdAt: new Date('2025-03-26T00:00:00.000Z'),
	startedAt: new Date('2025-03-26T00:00:00.001Z'),
	stoppedAt: new Date('2025-03-26T00:00:02.000Z'),
};

export const aiManualExecutionResponse: IExecutionResponse = {
	id: 'test-exec-id-2',
	finished: true,
	mode: 'manual',
	status: 'success',
	data: {
		resultData: {
			runData: {
				'AI Agent': [
					{
						executionStatus: 'success',
						startTime: +new Date('2025-03-30T00:00:00.002Z'),
						executionTime: 12,
						source: [],
						data: {},
					},
				],
				'AI Model': [
					{
						executionStatus: 'success',
						startTime: +new Date('2025-03-30T00:00:00.003Z'),
						executionTime: 3456,
						source: [],
						data: {
							ai_languageModel: [
								[
									{
										json: {
											tokenUsage: {
												completionTokens: 4,
												promptTokens: 5,
												totalTokens: 6,
											},
										},
									},
								],
							],
						},
					},
				],
			},
		},
	},
	workflowData: aiManualWorkflow,
	createdAt: new Date('2025-03-30T00:00:00.000Z'),
	startedAt: new Date('2025-03-30T00:00:00.001Z'),
	stoppedAt: new Date('2025-03-30T00:00:02.000Z'),
};
