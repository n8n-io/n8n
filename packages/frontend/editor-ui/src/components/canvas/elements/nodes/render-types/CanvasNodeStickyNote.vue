<script setup lang="ts">
/* eslint-disable vue/no-multiple-template-root */
import { useCanvasNode } from '@/composables/useCanvasNode';
import type { CanvasNodeStickyNoteRender } from '@/types';
import { ref, computed, useCssModule, onMounted, onBeforeUnmount } from 'vue';
import { NodeResizer } from '@vue-flow/node-resizer';
import type { OnResize } from '@vue-flow/node-resizer';
import type { XYPosition } from '@vue-flow/core';

defineOptions({
	inheritAttrs: false,
});

const emit = defineEmits<{
	update: [parameters: Record<string, unknown>];
	move: [position: XYPosition];
	activate: [id: string];
	deactivate: [id: string];
	'open:contextmenu': [event: MouseEvent];
}>();

const $style = useCssModule();

const { id, isSelected, isReadOnly, render, eventBus } = useCanvasNode();

const renderOptions = computed(() => render.value.options as CanvasNodeStickyNoteRender['options']);

const classes = computed(() => ({
	[$style.sticky]: true,
	[$style.selected]: isSelected.value,
	['sticky--active']: isActive.value, // Used to increase the z-index of the sticky note when editing
}));

/**
 * Resizing
 */

function onResize(event: OnResize) {
	emit('move', {
		x: event.params.x,
		y: event.params.y,
	});

	emit('update', {
		...(event.params.width ? { width: event.params.width } : {}),
		...(event.params.height ? { height: event.params.height } : {}),
	});
}

/**
 * Content change
 */

const isActive = ref(false);

function onInputChange(value: string) {
	emit('update', {
		content: value,
	});
}

function onSetActive(value: boolean) {
	if (isActive.value === value) return;

	isActive.value = value;

	if (value) {
		emit('activate', id.value);
	} else {
		emit('deactivate', id.value);
	}
}

function onActivate() {
	onSetActive(true);
}

/**
 * Context menu
 */

function openContextMenu(event: MouseEvent) {
	emit('open:contextmenu', event);
}

/**
 * Lifecycle
 */

onMounted(() => {
	eventBus.value?.on('update:node:activated', onActivate);
});

onBeforeUnmount(() => {
	eventBus.value?.off('update:node:activated', onActivate);
});
</script>
<template>
	<NodeResizer
		:min-height="80"
		:min-width="150"
		:height="renderOptions.height"
		:width="renderOptions.width"
		:is-visible="!isReadOnly"
		@resize="onResize"
	/>
	<N8nSticky
		v-bind="$attrs"
		:id="id"
		:class="classes"
		data-test-id="sticky"
		:height="renderOptions.height"
		:width="renderOptions.width"
		:model-value="renderOptions.content"
		:background-color="renderOptions.color"
		:edit-mode="isActive"
		:read-only="isReadOnly"
		@edit="onSetActive"
		@dblclick.stop="onActivate"
		@update:model-value="onInputChange"
		@contextmenu="openContextMenu"
	/>
</template>

<style lang="scss" module>
.sticky {
	position: relative;

	/**
	 * State classes
	 * The reverse order defines the priority in case multiple states are active
	 */

	&.selected {
		box-shadow: 0 0 0 4px var(--color-canvas-selected);
	}
}
</style>
