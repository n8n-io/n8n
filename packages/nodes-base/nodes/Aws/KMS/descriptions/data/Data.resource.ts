import type { INodeProperties } from 'n8n-workflow';

export const dataOperations: INodeProperties[] = [
	{
		displayName: 'Operation',
		name: 'operation',
		type: 'options',
		noDataExpression: true,
		displayOptions: {
			show: {
				resource: ['data'],
			},
		},
		options: [
			{
				name: 'Decrypt',
				value: 'decrypt',
				description: 'Decrypt ciphertext',
				action: 'Decrypt data',
				routing: {
					request: {
						method: 'POST',
						url: '/',
						headers: {
							'X-Amz-Target': 'TrentService.Decrypt',
							'Content-Type': 'application/x-amz-json-1.1',
						},
						body: {
							CiphertextBlob: '={{ $parameter["ciphertextBlob"] }}',
						},
					},
				},
			},
			{
				name: 'Encrypt',
				value: 'encrypt',
				description: 'Encrypt plaintext',
				action: 'Encrypt data',
				routing: {
					request: {
						method: 'POST',
						url: '/',
						headers: {
							'X-Amz-Target': 'TrentService.Encrypt',
							'Content-Type': 'application/x-amz-json-1.1',
						},
						body: {
							KeyId: '={{ $parameter["keyId"] }}',
							Plaintext: '={{ $parameter["plaintext"] }}',
						},
					},
				},
			},
			{
				name: 'Generate Data Key',
				value: 'generateDataKey',
				description: 'Generate a data encryption key',
				action: 'Generate data key',
				routing: {
					request: {
						method: 'POST',
						url: '/',
						headers: {
							'X-Amz-Target': 'TrentService.GenerateDataKey',
							'Content-Type': 'application/x-amz-json-1.1',
						},
						body: {
							KeyId: '={{ $parameter["keyId"] }}',
						},
					},
				},
			},
			{
				name: 'Re-Encrypt',
				value: 'reEncrypt',
				description: 'Re-encrypt data with a different key',
				action: 'Re-encrypt data',
				routing: {
					request: {
						method: 'POST',
						url: '/',
						headers: {
							'X-Amz-Target': 'TrentService.ReEncrypt',
							'Content-Type': 'application/x-amz-json-1.1',
						},
						body: {
							CiphertextBlob: '={{ $parameter["ciphertextBlob"] }}',
							DestinationKeyId: '={{ $parameter["destinationKeyId"] }}',
						},
					},
				},
			},
		],
		default: 'encrypt',
	},
];

export const dataFields: INodeProperties[] = [
	// Encrypt fields
	{
		displayName: 'Key ID',
		name: 'keyId',
		type: 'string',
		required: true,
		displayOptions: {
			show: {
				resource: ['data'],
				operation: ['encrypt', 'generateDataKey'],
			},
		},
		default: '',
		description: 'The ID or ARN of the KMS key',
	},
	{
		displayName: 'Plaintext',
		name: 'plaintext',
		type: 'string',
		required: true,
		displayOptions: {
			show: {
				resource: ['data'],
				operation: ['encrypt'],
			},
		},
		default: '',
		description: 'The plaintext data to encrypt (base64 encoded)',
	},
	// Decrypt fields
	{
		displayName: 'Ciphertext Blob',
		name: 'ciphertextBlob',
		type: 'string',
		required: true,
		displayOptions: {
			show: {
				resource: ['data'],
				operation: ['decrypt', 'reEncrypt'],
			},
		},
		default: '',
		description: 'The encrypted data (base64 encoded)',
	},
	// Re-encrypt fields
	{
		displayName: 'Destination Key ID',
		name: 'destinationKeyId',
		type: 'string',
		required: true,
		displayOptions: {
			show: {
				resource: ['data'],
				operation: ['reEncrypt'],
			},
		},
		default: '',
		description: 'The ID or ARN of the destination KMS key',
	},
	// Generate data key fields
	{
		displayName: 'Additional Fields',
		name: 'additionalFields',
		type: 'collection',
		placeholder: 'Add Field',
		default: {},
		displayOptions: {
			show: {
				resource: ['data'],
				operation: ['generateDataKey'],
			},
		},
		options: [
			{
				displayName: 'Key Spec',
				name: 'KeySpec',
				type: 'options',
				options: [
					{ name: 'AES 128', value: 'AES_128' },
					{ name: 'AES 256', value: 'AES_256' },
				],
				default: 'AES_256',
				description: 'Length of the data key',
			},
			{
				displayName: 'Number of Bytes',
				name: 'NumberOfBytes',
				type: 'number',
				default: 32,
				description: 'Length of the data key in bytes (1-1024)',
			},
		],
	},
	// Encrypt additional fields
	{
		displayName: 'Additional Fields',
		name: 'additionalFields',
		type: 'collection',
		placeholder: 'Add Field',
		default: {},
		displayOptions: {
			show: {
				resource: ['data'],
				operation: ['encrypt', 'decrypt'],
			},
		},
		options: [
			{
				displayName: 'Encryption Context (JSON)',
				name: 'EncryptionContext',
				type: 'json',
				default: '{}',
				description: 'Key-value pairs for additional authenticated data',
			},
		],
	},
];
