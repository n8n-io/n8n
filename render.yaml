services:
  - type: web
    name: luci-n8n-agent
    env: docker
    plan: free
    branch: master
    autoDeploy: true
    buildCommand: ""
    startCommand: "n8n start"
    envVars:
      - key: N8N_BASIC_AUTH_ACTIVE
        value: "true"
      - key: N8N_BASIC_AUTH_USER
        value: "admin"
      - key: N8N_BASIC_AUTH_PASSWORD
        value: "admin"
      - key: N8N_HOST
        value: "0.0.0.0"
      - key: N8N_PORT
        value: "5678"
      - key: WEBHOOK_URL
        value: https://luci-n8n-agent.onrender.com/
