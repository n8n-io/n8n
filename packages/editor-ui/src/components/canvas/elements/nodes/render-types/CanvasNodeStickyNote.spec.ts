import CanvasNodeStickyNote from '@/components/canvas/elements/nodes/render-types/CanvasNodeStickyNote.vue';
import { createComponentRenderer } from '@/__tests__/render';
import { createCanvasNodeProvide } from '@/__tests__/data';
import { createTestingPinia } from '@pinia/testing';
import { setActivePinia } from 'pinia';

const renderComponent = createComponentRenderer(CanvasNodeStickyNote);

beforeEach(() => {
	const pinia = createTestingPinia();
	setActivePinia(pinia);
});

describe('CanvasNodeStickyNote', () => {
	it('should render node correctly', () => {
		const { getByTestId } = renderComponent({
			global: {
				provide: {
					...createCanvasNodeProvide({
						id: 'sticky',
					}),
				},
			},
		});

		expect(getByTestId('canvas-sticky-note-node')).toMatchSnapshot();
	});
});
