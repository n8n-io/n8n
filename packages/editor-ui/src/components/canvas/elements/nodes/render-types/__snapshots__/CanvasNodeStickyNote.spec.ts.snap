// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CanvasNodeStickyNote > should render node correctly 1`] = `
<div
  class="n8n-sticky sticky clickable color-1 sticky"
  data-test-id="canvas-sticky-note-node"
  style="height: 180px; width: 240px;"
>
  <div
    class="wrapper"
  >
    <div
      class="n8n-markdown"
    >
      <div
        class="sticky"
      />
    </div>
  </div>
  <div
    class="sticky-textarea"
    style="display: none;"
  >
    <div
      class="el-textarea el-input--large n8n-input"
    >
      <!-- input -->
      
      <!-- textarea -->
      <textarea
        autocomplete="off"
        class="el-textarea__inner"
        name="sticky-input"
        placeholder=""
        rows="5"
        tabindex="0"
        title=""
      />
      <!--v-if-->
      
    </div>
  </div>
  <!--v-if-->
</div>
`;
